public/src/components/channelManagement/testEditorArticleCountEditor.tsx

```diff
@@ -62,7 +62,14 @@ const TestEditorArticleCountEditor: React.FC<TestEditorArticleCountEditorProps>
     periodInWeeks: articlesViewedSettings?.periodInWeeks.toString() || '',
   };
 
-  const { register, errors, handleSubmit } = useForm<FormData>({ mode: 'onChange', defaultValues });
+  const { register, errors, handleSubmit, reset } = useForm<FormData>({
+    mode: 'onChange',
+    defaultValues,
+  });
+
+  useEffect(() => {
+    reset(defaultValues);
+  }, [defaultValues.minViews, defaultValues.maxViews, defaultValues.periodInWeeks]);
```

Comment:
is `defaultValues` the current state of the form fields?

Reply:
it's the state when the form is first created - after a user starts interacting with the form react-hook-form handles the state internally

Reply:
ok, so this `useEffect` block will only execute once, at the start?

Reply:
yeah that's right! it should execute each time the user selects a new test (unless the test had identical settings)

Reply:
👍